### PR TITLE
Enable installing arbitrary simulator versions using xcode cli

### LINKED ISSUE #1310: [apple] Add ability to install a specific simulator runtime version with Xcode 16+
With Xcode 16 Apple is moving away from hosting newer simulator images publicly: https://developer.apple.com/documentation/xcode/installing-additional-simulator-runtimes#Install-and-manage-Simulator-runtimes-from-the-command-line

Instead for the following simulator versions:
- iOS 16+ (including iPadOS)
- tvOS 16+
- visionOS 1+
- watchOS 9+

they suggest downloading/installing simulator images through `xcodebuild` in the following way:
```
xcodebuild -downloadPlatform iOS -buildVersion <build-version>
```

The `<build-version>` can be acquired from: https://devimages-cdn.apple.com/downloads/xcode/simulators/index2.dvtdownloadableindex under `simulatorVersion` key (either `buildUpdate` or `version` value should work)

Example:
```
<dict>
	<key>authentication</key>
	<string>virtual</string>
	<key>category</key>
	<string>simulator</string>
	<key>contentType</key>
	<string>diskImage</string>
	<key>dictionaryVersion</key>
	<integer>2</integer>
	<key>fileSize</key>
	<integer>7336630623</integer>
	<key>identifier</key>
	<string>com.apple.dmg.iPhoneSimulatorSDK17_5</string>
	<key>name</key>
	<string>iOS 17.5 Simulator Runtime</string>
	<key>platform</key>
	<string>com.apple.platform.iphoneos</string>
	<key>simulatorVersion</key>
	<dict>
		<key>buildUpdate</key>
		<string>21F79</string>
		<key>version</key>
		<string>17.5</string>
	</dict>
	<key>source</key>
	<string>https://download.developer.apple.com/Developer_Tools/iOS_17.5_Simulator_Runtime/iOS_17.5_Simulator_Runtime.dmg</string>
	<key>version</key>
	<string>17.5.0.0</string>
</dict>
```

### Additional note

In the future, if/when we stop supporting older Xcode versions, we should remove code that manually downloads simulator images.

### PR DESCRIPTION
**Pull request by jkurdek**
https://github.com/dotnet/xharness/pull/1277 introduced ability to download and install simulator images using xcode cli. However, currently it will always default to the newest simulator version for the given platform. 

Xcode 16.1 added option to specify the simulator version when installing using xcode cli. This PR brings support for this into xharness.

Xcode 16.0 does not support specifying the simulator version, so a warning is issued to the user and the latest version is installed. 

Fixes #1310 

---

Scouting queues got upgraded to use latest versions, Xcode 16.1 included. Changes were tested on the scouting queue on Xcode 15.0, Xcode 16.0 and Xcode 16.1

### CODE REVIEW

**Review by rolfbjarne — APPROVED** [review]

File: src/Microsoft.DotNet.XHarness.CLI/Commands/Apple/Simulators/InstallCommand.cs [hunk 1]
```diff
@@ -121,57 +121,96 @@ protected override async Task<ExitCode> InvokeInternal(ILogger logger)
 
     private async Task<bool> Install(Simulator simulator)
     {
-        if (simulator.Source is null)
+        var rawXcodeVersion = await GetXcodeVersion();
+        var xcodeVersion = Version.Parse(rawXcodeVersion);
```
Comment (akoeplinger) [inline src/Microsoft.DotNet.XHarness.CLI/Commands/Apple/Simulators/InstallCommand.cs:125]: nit: all the callers of GetXcodeVersion() duplicate the Version.Parse so we could just return a `Version` from it directly.

**Review by akoeplinger — APPROVED** [review]

File: src/Microsoft.DotNet.XHarness.CLI/Commands/Apple/Simulators/SimulatorsCommand.cs [hunk 2]
```diff
@@ -180,7 +189,8 @@ protected async Task<IEnumerable<Simulator>> GetAvailableSimulators()
                 source: source,
                 installPrefix: installPrefix,
                 fileSize: (long)parsedFileSize,
-                isCryptexDiskImage: isCryptexDiskImage
+                isCryptexDiskImage: isCryptexDiskImage,
```
Comment (ivanpovazan) [inline src/Microsoft.DotNet.XHarness.CLI/Commands/Apple/Simulators/SimulatorsCommand.cs:192]: Do we still need to logic for `isCryptexDiskImage` ?

**Review by ivanpovazan — COMMENTED** [review]
We should test these changes on the CI more thoroughly, as currently we request a simulator version which is already installed, so unfortunately not much coverage is done. 
I added a https://github.com/dotnet/xharness/blob/main/tests/integration-tests/README.md some time ago explaining how to run tests on a scouting queue (Xcode 16) - we should focus on SimulatorCommands tests.

**Comment by jkurdek** [comment]
@ivanpovazan we are currently waiting for dnceng team to enable us to to use newer xcode versions on the scouting queues. Will improve the test once we have it:
https://github.com/dotnet/dnceng/issues/4379

**Comment by ivanpovazan** [comment]
Awesome! In the meantime we can use the queue to test Xcode 16, no?

File: src/Microsoft.DotNet.XHarness.CLI/Commands/Apple/Simulators/SimulatorsCommand.cs (hunk 2 above)
Comment (jkurdek) [reply src/Microsoft.DotNet.XHarness.CLI/Commands/Apple/Simulators/SimulatorsCommand.cs:192]: I think we can safely remove this now

**Comment by jkurdek** [comment]
@ivanpovazan I did run e2e tests on scouting queue installing simulators. I did check the following Xcode versions 15.0, 16.0 and 16.1 and the behaviour was correct. I tried installing versions requiring CLI download and those which can be downloaded using source.

**Comment by ivanpovazan** [comment]
Sounds good, thanks!

**Comment by ivanpovazan** [comment]
@jkurdek we should backport this to `release/9.0` and `release/8.0` branches

**Comment by jkurdek** [comment]
@ivanpovazan sure thing, will handle this

**Pull request merged**